tests/test_processor.py

```diff
@@ -1,5 +1,5 @@
-import socket
-import unittest
+from socket import socket, AF_INET, SOCK_STREAM
```

Comment:
I prefer `import <module>` than `from <module> import <function>`. In my opinion using regular import improves readability. When reviewing Python code, I like seeing where the given function or class comes from right where it is used. It saves me from scrolling to the top of the module to get that info.